smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/impl/KafkaSink.java

```diff
@@ -158,6 +168,47 @@ private void handleWriteResult(AsyncResult<?> ar, Message<?> message, ProducerRe
                 kafkaHeaders);
     }
 
+    private void createOutgoingTrace(Message<?> message, String topic, int partition, Headers headers) {
+        if (configuration.getTracingEnabled()) {
+            Optional<TracingMetadata> tracingMetadata = TracingMetadata.fromMessage(message);
+
+            final Span.Builder spanBuilder = TRACER.spanBuilder(topic)
```

Comment:
Why?

That's not what is recommended by current OpenTelemetry docs

Reply:
Where did you find the guidelines? 

The operation name should be low cardinality which topic name certainly is, but it should represent the operation wich in our case is either consume or produce.

Reply:
We should also add the topic name to tags. This will help with the data semantics and in aggregation jobs since we cannot rely on the operation name.

Reply:
https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/messaging.md looks like it has been updated since last I saw, previously it only mentioned topic name.

Topic name is added to `messaging.destination`

Reply:
It might be the important fact here is to keep it low cardinality and then describe what it does. `produce` would be also good but including the topic name is even better.